File: 1281_Subtract_the_Product_and_Sum_of_Digits_of_an_Integer/solution.go
// https://leetcode.com/problems/subtract-the-product-and-sum-of-digits-of-an-integer
package leetcode1281
/*
	
 */

func subtractProductAndSum(n int) int {
		product:=-1
		sum:=0

		for n > 0{
			digit := n %10
			if product == -1{
				product = digit
			}else{
				product *= digit
			}
			sum += digit

			n = n /10
		}

		return product - sum
}

File: 1281_Subtract_the_Product_and_Sum_of_Digits_of_an_Integer/solution.cpp
/*
1281. Subtract the Product and Sum of Digits of an Integer

Level: Easy

https://leetcode.com/problems/subtract-the-product-and-sum-of-digits-of-an-integer
*/
#include <vector>
#include <string>
#include <unordered_set>

#include <algorithm>

#include "common/ListNode.h"
#include "common/TreeNode.h"

using namespace std;

/*
  Solution: 
*/

class Solution
{
public:
  int subtractProductAndSum(int n)
  {
    int product = 1;
    int sum = 0;
    while (n > 0)
    {
      int digit = n % 10;
      product *= digit;
      sum += digit;
      n = n / 10;
    }
    return product - sum;
  }
};